src/features/search/components/SearchResults.tsx

```diff
@@ -123,11 +127,32 @@ export const SearchResults: React.FC = () => {
     [searchState.query]
   )
 
-  const ListHeaderComponent = useMemo(
-    () => <NumberOfResults nbHits={nbHits} />,
+  const ListHeaderComponent = useMemo(() => {
+    const shouldDisplayGeolocationButton =
+      position === null &&
+      !params?.offerTypes?.isDigital &&
+      params?.offerCategories?.[0] !== SearchGroupNameEnumv2.EVENEMENTS_EN_LIGNE &&
+      params?.offerCategories?.[0] !== SearchGroupNameEnumv2.PLATEFORMES_EN_LIGNE
+
+    return (
+      <React.Fragment>
+        <NumberOfResults nbHits={nbHits} />
+        {!!shouldDisplayGeolocationButton && (
+          <GeolocationButtonContainer>
+            <GeolocationButton onPress={() => showGeolocPermissionModal()} />
```

Comment:
```suggestion
            <GeolocationButton onPress={showGeolocPermissionModal} />
```
Ca évite de recréer une fonction à chaque render, donc ça optimise les perfs. En plus, c'est inutile de recréer :P